Slime Mall/Assets/Scripts/Player/PlayerController.cs

```diff
@@ -104,28 +94,21 @@ void Update()
                 //In range
                 if(Vector3.Distance(transform.position, targetNPC.transform.position) <= interactRadius)
                 {
-                    animator.SetTrigger("Consume");
-                    agent.enabled = false;
-                    //anim?
-                    transform.position = targetNPC.transform.position;
-                    NPCManager.Instance.KillNPC(targetNPC);
-                    GameManager.Instance.UpdateScore();
-                    agent.enabled = true;
-                    targetNPC = null;
+                    KillNPC();
                 }
             }
 
-            else if(targetHS)
+            //Not hidden and has HS
+            else if((targetHS && !bIsHidden) || bIsTeleporting)
```

Comment:
What is the targetHS variabable?

Reply:
HS means Hiding Spot, targetX refers to the current stored object of type X